services/audit-service/src/main/java/datawave/microservice/audit/health/rabbit/RabbitHealthChecker.java

```diff
@@ -0,0 +1,705 @@
+                    log.trace("Fixing invalid binding: [" + bindings.desired + "]");
+                    
+                    if (rabbitClient != null) {
+                        try {
+                            // remove invalid bindings
+                            if (bindings.detected.getDestinationType().equals(Binding.DestinationType.EXCHANGE))
+                                rabbitClient.unbindExchange("/", bindings.detected.getDestination(), bindings.detected.getExchange(),
+                                                bindings.detected.getRoutingKey());
+                            else if (bindings.desired.getDestinationType().equals(Binding.DestinationType.QUEUE))
+                                rabbitClient.unbindQueue("/", bindings.detected.getDestination(), bindings.detected.getExchange(),
+                                                bindings.detected.getRoutingKey());
+                            
+                            // add correct bindings
+                            if (bindings.desired.getDestinationType().equals(Binding.DestinationType.EXCHANGE))
+                                rabbitClient.bindExchange("/", bindings.desired.getDestination(), bindings.desired.getExchange(),
+                                                bindings.desired.getRoutingKey(), bindings.desired.getArguments());
+                            else if (bindings.desired.getDestinationType().equals(Binding.DestinationType.QUEUE))
+                                rabbitClient.bindQueue("/", bindings.desired.getDestination(), bindings.desired.getExchange(), bindings.desired.getRoutingKey(),
+                                                bindings.desired.getArguments());
+                        } catch (Exception e) {
+                            log.trace("Unable to recreate invalid binding", e);
+                        }
+                    }
+                }
+            }
+            log.trace("RabbitMQ Recovery - Complete");
+        }
+    }
+    
+    /**
+     * Used to determine whether RabbitMQ is healthy.
+     * <p>
+     * This method is synchronized with the runHealthCheck method to ensure that the health check is given time to complete before deciding whether to accept an
+     * audit request via the {@link AuditController}
+     * 
+     * @return true if RabbitMQ is healthy, false if RabbitMQ is unhealthy
+     */
+    @Override
+    public boolean isHealthy() {
+        synchronized (this) {
+            return clusterHealthy && exchangesHealthy && queuesHealthy && bindingsHealthy;
+        }
+    }
+    
+    private void updateOutageStats(RabbitOutageStats outageStats) {
+        if (numNodesMissing > 0)
+            outageStats.setNumNodesMissing(Math.max(outageStats.getNumNodesMissing(), numNodesMissing));
+        
+        outageStats.getMissingExchanges().addAll(missingExchanges.stream().map(Exchange::getName).collect(Collectors.toList()));
```

Comment:
Not that this gets called a lot, so it's not a big deal, but you could avoid creating an extra list by doing something like:
```
missingExchanges.stream().map(Exchange::getName).forEach(e -> outageStats.getMissingExchanges().add(e));
```

Reply:
Good point.  Updated!